src/Controller/TaskController.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace App\Controller;
+
+use App\Entity\Task;
+use App\Form\TaskType;
+use App\Repository\TaskRepository;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Attribute\Route;
+
+class TaskController extends AbstractController
+{
+
+    public function __construct(private EntityManagerInterface $em)
+    {
+    }
+
+    #[Route('/task', name: 'app_task',  methods: ["GET"])]
+    public function index(TaskRepository $taskrepo): Response
+    {
+        $this->denyAccessUnlessGranted("ROLE_USER");
+
+        $tasks = $taskrepo->findAll();
+
+        return $this->render('task/task.html.twig', [
+            'tasks' => $tasks
+        ]);
+    }
+
+    #[Route('/task/edit/{id}', name: 'app_task_edit',  methods: ["GET", "POST"])]
+    public function edit(Task $task, Request $request): Response
+    {
+        $this->denyAccessUnlessGranted("ROLE_USER");
+
+        $form = $this->createForm(TaskType::class, $task);
+        $form->handleRequest($request);
+        if ($form->isSubmitted() && $form->isValid()) {
+            $data = $form->getData();
+            $task->setTaskname($data->getTaskname());
+            $task->setDescription($data->getDescription());
+            $task->setPoints($data->getPoints());
+            $this->em->persist($task);
+            $this->em->flush();
+            $this->addFlash('success', 'Tâche modifiée');
+
+            return $this->redirectToRoute('app_task');
+        }
```

Comment:
Tu n'as pas besoin de reconstituer le form, Symfony s'en occupe tant que tu as fourni $task ici :  `$this->createForm(TaskType::class, $task)` et que le formulaire est valide.

```suggestion
        $form = $this->createForm(TaskType::class, $task);
        $form->handleRequest($request);
        if ($form->isSubmitted() && $form->isValid()) {
            $this->em->persist($task);
            $this->em->flush();
            $this->addFlash('success', 'Tâche modifiée');

            return $this->redirectToRoute('app_task');
        }
```

Reply:
Du coup je lui ai donné $task qui est une instance de ma class Task. Elle récupère l'ID dans l'argument pour récupérer la bonne task. Ca c'est la magie noire de Symfony ?
Donc ensuite, elle affiche directement dans les inputs les informations récupérées, la où une création de task va afficher un input vide en attendant d'être rempli. 
J'ai bien compris ?